src/multidim/setdiff.jl

```diff
@@ -105,3 +105,25 @@ end
 #
 #     return results_list
 # end
+
+import IntervalArithmetic.hull
+doc"""
+    hull(a, b)
+
+Returns the `box hull` of the IntervalBoxes `a` and `b`, considered as
+(extended) $N$ dimensional sets of real numbers, i.e. the smallest region that contains
+all of `a` and `b`.
+"""
+function hull(A::IntervalBox{N,T}, B::IntervalBox{N,T}) where {N,T}
+    result_list = Vector{Interval{T}}()
```

Comment:
Here you are creating an intermediate array that it turns out is actually unnecessary. There is a function `ntuple` that makes a tuple out of a function that you can use here. This is used in a few places in the `IntervalBox` code.